js/map.js

```diff
@@ -54,13 +59,42 @@ function generateMonsters(){
     monsters = [];
     let numMonsters = level+1;
     for(let i=0;i<numMonsters;i++){
-        spawnMonster();
+        spawnMonster(level);
     }
 }
 
-function spawnMonster(){
+function spawnMonster(level){
     // let monsterType = shuffle([Goose, Ant, Mushroom, Eater, Toast])[0];
-    let monsterType = shuffle([Rock, Paper, Scissors ,Rock_Plus, Paper_Plus, Scissors_Plus, Rock_Anti, Paper_Anti, Scissors_Anti])[0];
+
+     let monsterType = shuffle(chooseMonsterType(level))[0];
     let monster = new monsterType(randomPassableTile());
     monsters.push(monster);
 }
+
+function chooseMonsterType(level){  
+
+  if(level < 4){    
+    return normalMonster();
+  }
+  else if(level < 7){
+    return[antiMonster()[0],...normalMonster()]
+  } else{
+    return[antiMonster()[0],normalMonster()[0],...plusMonster()]
+  }
+}
+
+function normalMonster(){
```

Comment:
You can make one function
`shuffleMonster(arg)` arg can ben `[NORMAL,PLUS,ANTI]`
Instead of having 3 different functions.

Reply:
Like, unifying `function normalMonster()` , `function plusMonster()` and  `function antiMonster()` ?

Reply:
@abnersajr  what would be the difference between calling `shuffleMonster(arg)` and `shuffle(monster_kind)`  ?

Reply:
huuumm.... there is a extra layer of randomness I guess, but i think it would be better with fewer functions... I'll give it a try, thanks.

Reply:
@negreirosleo shuffle can be used to any kind of array. The suffix monster scopes this `shuffle` to a scope of monsters. It's a suggestion based on this new code. Not a mandatory change.
And you can add to the scope of `shuffleMonsters` some kind of validation, default values and so on.
Like I've said this isn't a mandatory change. Just sharing another POV.